src/test/scala/firrtlTests/transforms/TopWiringTest.scala

```diff
@@ -0,0 +1,167 @@
+// See LICENSE for license details.
+
+package firrtlTests
+package transform
+
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import org.scalatest.junit.JUnitRunner
+import scala.io.Source
+import java.io._
+
+import firrtl._
+import firrtl.ir.{Circuit, Type, GroundType, IntWidth}
+import firrtl.Parser
+import firrtl.passes.PassExceptions
+import firrtl.annotations.{
+   Named,
+   CircuitName,
+   ModuleName,
+   ComponentName,
+   Annotation
+}
+import firrtl.transform.TopWiring._
+
+
+/**
+ * Tests TopWiring transformation
+ */
+class TopWiringTests extends LowTransformSpec {
+
+   def TopWiringDummyOutputFilesFunction(prefix: String, dir: String, mapping: Seq[((ComponentName, Type, Boolean, Seq[String]), Int)], state: CircuitState): CircuitState = {
+     state
+   }
+
+   def TopWiringTestOutputFilesFunction(prefix: String, dir: String, mapping: Seq[((ComponentName, Type, Boolean, Seq[String]), Int)], state: CircuitState): CircuitState = {
+     val testOutputFile = new PrintWriter(new File(dir, "TopWiringOutputTest.txt" ))
+     mapping map {
+          case ((_, tpe, _, path), index) => {
+            val portwidth = tpe match { case GroundType(IntWidth(w)) => w }
```

Comment:
This will error out if you try to match on a non-GroundType. Is this guaranteed to never happen?

Reply:
Does midForm have non-GroundTypes?

Reply:
MidForm can have aggregates. `LowerTypes` is what removes those.